### ISSUE TITLE
`exc_info` not available when running jobs with `is_async=False`

### ISSUE THREAD

**Issue by aparcar** [rq/rq#1633]
I'm running jobs that may raise exceptions which would be forwarded to the user via `job.exc_info`. I wanted to test this by setting the queue to sync (aka `is_async=False`). Now some jobs intentionally fail, however the `exc_info` is not set. It stays on `None` while the job status is set to *failed*.

Is this intended behavior? If not I'll look into fixing this.

**Comment by selwin** [comment]
Sorry for the late response, no definitely not the intended behavior ;)

**Comment by dhvcc** [comment]
+1 on this one. Recently tried rq and was super consufed to see empty `exc_info`. I'm using `django_rq` for reference (I think this is the same issue here)

**Comment by dhvcc** [comment]
Upon further investigation I can see that I don't get any `exc_info` no matter if the queue is async or not. I have a function

```python
def sleeps_and_raises(t, exc):
    sleep(t)
    raise exc
```

Doesn't matter if I queue it manually creating my own `Queue` object (async or not) and whether I'm running a worker through `django_rq`'s `./manage.py rqworker default` or with ` rq worker default` (except latter fails tasks with django setup error). I used `@job` decorator and without it, exc_info is **always** `None` while `is_failed` is `True`.

This is how my default queue looks like (remember, I created manual queues with `is_async=False` and the behaviour was stil lthe same)

```python
╭──────────────────────────── <class 'django_rq.queues.DjangoRQ'> ────────────────────────────╮
│ A subclass of RQ's QUEUE that allows jobs to be stored temporarily to be                    │
│ enqueued later at the end of Django's request/response cycle.                               │
│                                                                                             │
│ ╭─────────────────────────────────────────────────────────────────────────────────────────╮ │
│ │ DjangoRQ('default')                                                                     │ │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                             │
│        canceled_job_registry = <rq.registry.CanceledJobRegistry object at 0x7f941d4556c0>   │
│                   connection = Redis<ConnectionPool<Connection<host=localhost,port=8883,db… │
│                        count = 0                                                            │
│              DEFAULT_TIMEOUT = 180                                                          │
│        deferred_job_registry = <rq.registry.DeferredJobRegistry object at 0x7f941d373d30>   │
│          failed_job_registry = <rq.registry.FailedJobRegistry object at 0x7f941d4569e0>     │
│        finished_job_registry = <rq.registry.FinishedJobRegistry object at 0x7f941d4541c0>   │
│                     is_async = True                                                         │
│                      job_ids = []                                                           │
│                         jobs = []                                                           │
│                          key = 'rq:queue:default'                                           │
│                         name = 'default'                                                    │
│ redis_queue_namespace_prefix = 'rq:queue:'                                                  │
│            redis_queues_keys = 'rq:queues'                                                  │
│         redis_server_version = None                                                         │
│        registry_cleaning_key = 'rq:clean_registries:default'                                │
│       scheduled_job_registry = <rq.registry.ScheduledJobRegistry object at 0x7f941d456e00>  │
│         started_job_registry = <rq.registry.StartedJobRegistry object at 0x7f941d456d70>    │
╰─────────────────────────────────────────────────────────────────────────────────────────────╯

```

**Comment by lowercase00** [comment]
@dhvcc mind sharing your setup and versions (redis, rq, redis-py), so we can try to reproduce this please?

**Comment by lowercase00** [comment]
Fixed by #1799
